docs: Add multi-lang module structure page

vikram-dagger opened the pull request.

vikram-dagger commented:
@helderco I've created this as a separate multi-language page but on review, it seems like this is specific enough to each SDK that it should actually be part of the language-specific page (#7408). Let me know if you agree.

vikram-dagger commented:
This was a difficult page to merge, because the content was significantly different for each languages. I needed to backfill some content and also adjust the structure of the sections. To confirm there are no errors, I would like review and approval from experts in our team. Pinging:

- @vito @sipsma @jedevc for Go
- @helderco for Python
- @TomChv for TypeScript

When reviewing, if you see areas where we can modify or optimize the internal page structure further, please add comments.

TomChv commented on docs/current_docs/manuals/developer/module-structure.mdx:
```diff
@@ -0,0 +1,455 @@
+</TabItem>
+<TabItem value="TypeScript">
+
+The runtime container is currently hardcoded to run in Node.js 21.3 (although this may be configurable in future).
```
This will change soon when we will use LTS version, see #7433

TomChv reviewed (commented):
Thanks for these changes, I added some update suggestions based on new changes that were introduced with #7096.

I like the overall structure, you found a nice balance between generic information and language specific.

vikram-dagger commented on docs/current_docs/manuals/developer/module-structure.mdx:
```diff
@@ -0,0 +1,455 @@
+  "dependencies": {
+    "typescript": "^5.3.2"
+  },
+  "devDependencies": {
+    "@dagger.io/dagger": "./sdk"
+  }
+}
```
Added

vikram-dagger commented on docs/current_docs/manuals/developer/module-structure.mdx:
````diff
@@ -0,0 +1,455 @@
+
+```shell
+# executed by the runtime container
+npm install
````
Added

vikram-dagger commented on docs/current_docs/manuals/developer/module-structure.mdx:
````diff
@@ -0,0 +1,455 @@
+
+```shell
+# executed by the runtime container
+npm install --package-lock-only ./sdk
````
Added, please check if correct

vikram-dagger commented:
> Thanks for these changes, I added some update suggestions based on new changes that were introduced with #7096.

I've added these changes, can you please review once again @TomChv ? Thanks!

vikram-dagger commented:
Ping @vito @sipsma @jedevc @helderco @TomChv for review, see https://github.com/dagger/dagger/pull/7411#issuecomment-2122665088

kpenfound reviewed (approved):
One change in comment
